helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java

```diff
@@ -442,6 +442,29 @@ public Response getClusterTopology(@PathParam("clusterId") String clusterId) thr
     return OK(objectMapper.writeValueAsString(clusterTopology));
   }
 
+  @ResponseMetered(name = HttpConstants.READ_REQUEST)
+  @Timed(name = HttpConstants.READ_REQUEST)
+  @GET
+  @Path("{clusterId}/topologymap")
+  public Response getClusterTopologyMap(@PathParam("clusterId") String clusterId) {
+    if (!doesClusterExist(clusterId)) {
+      return notFound(String.format("Cluster %s does not exist", clusterId));
+    }
+    HelixAdmin admin = getHelixAdmin();
+    Map<String, List<String>> topologyMap = admin.getClusterTopology(clusterId).getTopologyMap();
+    return JSONRepresentation(topologyMap);
+  }
+
+  @ResponseMetered(name = HttpConstants.READ_REQUEST)
+  @Timed(name = HttpConstants.READ_REQUEST)
+  @GET
+  @Path("{clusterId}/faultzonemap")
```

Comment:
From the java code logic, it seems these 2 REST APIs actually return a topology which is represented by a map.
If so, I think the REST endpoint `{clusterId}/topology` is good and clear enough? We don't need the `map` to specify the data format.

And for fault zone, how about adding a query parameter `domain_type` based on the `/topology` endpoint: `/topology?domain_type=fault_zone`?  (correct me if I am not clear about it, as I just understand it by reading the code) Then it looks cleaner and is easier to read.

Reply:
There is already a legacy `topology` REST API in this class, so we have to use a different one. Any other idea than `topologymap`?
Regarding the suggestion of fault zone query, what you said is actually my previous implementation. However, we then decided to only expose the most common use case, i.e., return the topology under fault zone as an API to our users. So this fault zone API is not a generic one that accepts any level of domain. Instead, it's a fixed one only for "fault zone type" defined in topology.